src/main/kotlin/org/rust/lang/core/completion/RsLintCompletionProvider.kt

```diff
@@ -0,0 +1,545 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.ElementPattern
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.parentOfType
+import com.intellij.util.ProcessingContext
+import org.rust.ide.icons.RsIcons
+import org.rust.ide.icons.multiple
+import org.rust.lang.RsLanguage
+import org.rust.lang.core.RsPsiPattern
+import org.rust.lang.core.psi.RsMetaItem
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.ext.qualifier
+import org.rust.lang.core.psiElement
+import javax.swing.Icon
+
+object RsLintCompletionProvider : RsCompletionProvider() {
+    private const val RUSTC_SPECIFIC_PRIORITY = 5.0
+    private const val RUSTC_GROUP_PRIORITY = 4.0
+    private const val CLIPPY_PRIORITY = 3.0
+
+    private val GROUP_ICON = RsIcons.ATTRIBUTE.multiple()
+
+    override fun addCompletions(parameters: CompletionParameters,
+                                context: ProcessingContext,
+                                result: CompletionResultSet) {
+        val path = parameters.position.parentOfType<RsPath>() ?: return
+        val prefix = getPathPrefix(path)
+
+        LINTS.forEach {
+            val name = it.name
+            if (!name.startsWith(prefix)) return@forEach
+
+            val offered = name.removePrefix(prefix)
+            val element = LookupElementBuilder.create(offered)
+                .withIcon(getIcon(it))
+                .withPriority(getPriority(it))
+            result.addElement(element)
+        }
+    }
+
+    override val elementPattern: ElementPattern<out PsiElement>
+        get() {
+            return PlatformPatterns.psiElement()
+                .withLanguage(RsLanguage)
+                .withParent(RsPath::class.java)
+                .inside(
+                    psiElement<RsMetaItem>().withSuperParent(2, RsPsiPattern.lintAttributeMetaItem)
+                )
+        }
+
+    private fun getIcon(lint: Lint): Icon = when (lint) {
```

Comment:
If we decide to split the current completion provider into two ones, it may be convenient to move this method into `Lint` itself